### PR TITLE
feat: add electron web demo

### PR DESCRIPTION
**Pull request by DominicGBauer**
## Description
Add a demo showcasing how Electron can be used with a rendered web application.

## Work Done
* Included electron demo application
* Updated root README

## How to test
* Run `cd demos/example-electron`
* Run `pnpm start`
* Confirm app works as expected when offline or online and switching between offline to online

## Screenshot
image

### CODE REVIEW

**Comment by Chriztiaan** [comment]
Tested app. Works!

File: demos/electron-supabase-todolist/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# PowerSync + Supabase Electron with rendered Web App Demo: Todo List
+
+## Note: Beta Release
+
+This package is currently in a beta release.
```
Comment (benitav) [inline demos/electron-supabase-todolist/README.md:5]:
```suggestion
The [web SDK](/packages/web/) is currently in a beta release.
```

**Comment by rkazakov** [comment]
Thanks guys. Looking forward to proper Electron apps support.

**Review by stevensJourney — COMMENTED** [review]
This does showcase how the PowerSync client can be used inside the Electron renderer, but I am not a fan of duplicating the React Todolist demo's source in this demo. It should be possible to import that demo here instead. Were there any issues with that approach?

**Comment by DominicGBauer** [comment]
There were some tweaks I needed to add that does change this a bit. Also needed to make some minor UI changes since there is no browser back button.

File: demos/example-electron/package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "example-electron",
+  "productName": "example-electron",
+  "version": "1.0.0",
+  "description": "Example of using Electron with PowerSync",
+  "main": ".vite/build/main.js",
```
Comment (stevensJourney) [inline demos/example-electron/package.json:6]: We maybe should add `private: true` here?

File: demos/react-supabase-todolist/src/app/auth/register/page.tsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
+import { useNavigate } from 'react-router-dom';
 import { useSupabase } from '@/components/providers/SystemProvider';
 import { LoginDetailsWidget } from '@/components/widgets/LoginDetailsWidget';
-import { DEFAULT_ENTRY_ROUTE, LOGIN_ROUTE } from '@/app/router';
```
Comment (stevensJourney) [inline demos/react-supabase-todolist/src/app/auth/register/page.tsx:4]: Would be nice to still use the `@` here

**Review by stevensJourney — APPROVED** [review]
I'm happy with this example. Excellent work!

**Pull request merged**